### PR TITLE
RHCLOUD-35053: Relate resources and workspaces to default (not root)

### PR DESCRIPTION
**Pull request by alechenninger**
## Link(s) to Jira
- https://issues.redhat.com/browse/RHCLOUD-35053

## Description of Intent of Change(s)

Everything should go to/under the "default" workspace in order to maintain equivalent inheritance behavior, while at the same time allow for escaping inheritance with new workspaces under the root without having to move resources around. This PR changes migration logic to follow this.

As a result, dual write is also decoupled from the root workspace ID.

## Local Testing

Unit test updates. More tests to come in follow up PR.

## Checklist
- [x] if API spec changes are required, is the spec updated?
- [x] are there any pre/post merge actions required? if so, document here.
- [x] are theses changes covered by unit tests?
- [x] if warranted, are documentation changes accounted for?
- [x] does this require migration changes?
  - [x] if yes, are they backwards compatible?
- [x] is there known, direct impact to dependent teams/components?
  - [x] if yes, how will this be handled?

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Practices Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Review by astrozzc — APPROVED** [review]
LGTM!

**Comment by alechenninger** [comment]
TY! I'll merge after #1184 is merged in (since they build on one another)

**Review by lpichler — APPROVED** [review]

**Comment by alechenninger** [comment]
Cherry-picked on top of latest dual writes branch and confirmed diffs are the same

**Pull request merged**
